modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py

```diff
@@ -269,12 +268,44 @@ def run_test_connection_stage(args, extended_node_dict, upgrade_status_service):
             print(f"### Test connection to all DSF nodes succeeded")
 
 
-def run_collect_python_location_step(args, extended_node_dict, upgrade_status_service):
-    """
-    Collects the Python location in the DSF nodes and fills it in the extended_node_dict
-    """
-    if should_run_python(args):
-        collect_python_locations(extended_node_dict, args.stop_on_failure, upgrade_status_service)
+def run_collect_facts_step(args, extended_node_dict, upgrade_status_service):
+
+    for extended_node in extended_node_dict.values():
+        if upgrade_status_service.should_collect_facts(extended_node.get('dsf_node_id')):
+            upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                         UpgradeStatus.RUNNING_COLLECT_FACTS)
+            try:
+                collect_facts(extended_node)
+                upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                             UpgradeStatus.COLLECT_FACTS_SUCCEEDED)
+            except Exception as e:
+                upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                             UpgradeStatus.COLLECT_FACTS_FAILED)
+                if args.stop_on_failure:
+                    raise UpgradeException(
+                        f"Collecting Python location in {extended_node.get('dsf_node_name')} failed") from e
+                return None
+
+
+def collect_facts(extended_node):
+
+    with remote_client_context(extended_node.get('dsf_node'), _connection_timeout) as client:
+        collect_sysconfig(extended_node, client)
+        collect_python_location(extended_node)
+
+
+def collect_sysconfig(extended_node, client):
```

Comment:
It is better not to mix up the "sonar" logic with the general upgrade flow logic (its steps).
Please move this logic to some sonar_utils.py file that will contain the "sonar" logic, that will only return a map if the sysconfig settings (i.e. without changing the extended_node structure). In additional, I think it is better approach to not save "sysconfig" on the extended node, but to save only the info we need from it, as you did with extended_node['python_location'] and the rest of the code uses "python_location" and not familiar with its key in the sysconfig file. Just to mention that in case of adding a field to the extended_node, then it suppose to be initialized as undefined in the crate_extended_node func so the extended_node structure will be well defined.

Reply:
If collecting this config is no longer to be a "step", then why was getting the python location a step previously? This does more, and especially now the python location just uses this data to get the path.

I'm not sure this PR is the place to refactor how you set up steps and where functions are located, I'm just trying to get thing set up currently so that I can actually run the preflight/postflight that is bundled with sonar.